docs/HowSwiftImportsCModules.md

```diff
@@ -0,0 +1,922 @@
+| `char8_t` (proposed for C++20) | Not mapped |
+| `char16_t`                           | `typealias CChar16 = UInt16` |
+| `char32_t`                           | `typealias CChar32 = Unicode.Scalar` |
+| `float`                              | `typealias CFloat = Float` |
+| `double`                             | `typealias CDouble = Double` |
+| `long double`                        | `CLongDouble`, which is a typealias to `Float80` or `Double`, depending on the platform.<br>There is no support for 128-bit floating point. |
+
+First of all, notice that C types are mapped to Swift typealiases, not directly
+to the underlying Swift types. The names of the typealiases are based on the
+original C types. These typealiases allow developers to easily write Swift code
+and APIs that work with APIs and data imported from C without a lot of `#if`
+conditions.
+
+This table is generally unsurprising: C types are mapped to corresponding
+explicitly-sized Swift types, except for the C `long`, which is mapped to `Int`
+on 32-bit and 64-bit LP64 platforms. This was done to enhance portability
+between 32-bit and 64-bit code.
+
+The difficulty with the C `long` type is that it can be 32-bit or 64-bit
+depending on the platform. If C `long` was mapped to an explicitly-sized Swift
+type, it would map to different Swift types on different platforms, making it
+more difficult to write portable Swift code (the user would have to compile
+Swift code for both platforms to see all errors; fixing compilation errors on
+one platform can break another platform.) By mapping `long` a distinct type,
+`Int`, the language forces the user to think about both cases when compiling for
+either platform.
+
+Nevertheless, mapping C `long` to `Int` does not work universally.
+Specifically, it does not work on LLP64 platforms (for example, Windows
+x86\_64), where C `long` is 32-bit and Swift's `Int` is 64-bit. On LLP64
+platforms, C `long` is mapped to Swift's explicitly-sized `Int32`.
+
+Typedefs for integer types in the C standard library are mapped like this (from
+[`swift.git/swift/lib/ClangImporter/MappedTypes.def`](../lib/ClangImporter/MappedTypes.def):
+
+| C and C++ types  | Swift types |
+| ---------------- | ----------- |
+| `uint8_t`        | `UInt8`     |
+| `uint16_t`       | `UInt16`    |
+| `uint32_t`       | `UInt32`    |
+| `uint64_t`       | `UInt64`    |
+| `int8_t`         | `Int8`      |
+| `int16_t`        | `Int16`     |
+| `int32_t`        | `Int32`     |
+| `int64_t`        | `Int64`     |
+| `intptr_t`       | `Int`       |
+| `uintptr_t`      | `UInt`      |
+| `ptrdiff_t`      | `Int`       |
+| `size_t`         | `Int`       |
+| `rsize_t`        | `Int`       |
+| `ssize_t`        | `Int`       |
+
+```c
+// C header.
+
+double Add(int x, long y);
+```
+
+```swift
+// C header imported in Swift.
+
+func Add(_ x: CInt, _ y: CLong) -> CDouble
```

Comment:
I wish it did use the typealiases, but the generated interface in Xcode is actually:

```swift
func Add(_ x: Int32, _ y: Int) -> Double
```

Reply:
Indeed, seems like the either ClangImporter or the module interface printer resolves the typealiases... In any case, the implementation of ClangImporter does use the underlying typealiases in the standard library.